docs/en/cookbook/generated-columns.rst

```diff
@@ -0,0 +1,46 @@
+Generated Columns
+=================
+
+Generated columns, sometimes also called virtual columns, are populated by
+the database engine itself. They are a tool for performance optimization, to
+avoid calculating a value on each query.
+
+You can define generated columns on entities and have Doctrine map the values
+to your entity.
+
+Declaring a generated column
+----------------------------
+
+There is no explicit mapping instruction for generated columns. Instead, you
+specify that the column should not be written to, and define a custom column
+definition.
+
+.. code-block:: php
+    .. literalinclude:: generated-columns/Person.php
+
+* ``insertable``, ``updatable``: Setting these to false tells Doctrine to never
+  write this column - writing to a generated column would result in an error
+  from the database.
+* ``columnDefinition``: The instructions on how to generate the column can not be
+  expressed with Doctrine mapping. ``columnDefinition`` specifies the full SQL
+  to create the column. To allow to use database specific features, this
+  attribute does not use DQL. Note that we use the database column names with
+  snake_case to reference the fields, and not the entity fields with camelCase.
```

Comment:
This makes a lot of assumptions on configuration and is also not very clear in some parts.  Can you re-read this a few times and simplify?

Snake Case vs Camel Case depends on how the naming strategy is configured. Default behavior is not the Symfony behavior.

Reply:
uh, i was not aware that snake_case is not the default strategy. i changed the example and adjusted the text to explain things without assumptions. better?